### DISCUSSION TITLE
Tts api

### DISCUSSION THREAD

**Discussion by MattL0** [mikebrady/shairport-sync#1562] [Ideas]
Would this be possible?

**Comment by noelhibbard** [comment]
I use TTS at my house for announcements so I'll share what I've come up with. I take advantage of PulseAudio to group all the rooms that I want to send announcements to and also take advantage of the duck module. So if music is playing in any of these rooms it will suck the volume and then play the announcements. For the TTS part of things I'm taking advantage of Google's cloud speech API's and also cache the mp4 output to disk using the sha hash of the text string as the file name. So when I make an announcement I first do a sha hash of the string and then look to see if I have it in my cache before calling google to do the work. I use MQTT for submitting requests to the program so I named it speech2mqtt. I think leveraging shairport-sync is beyond the scope of the project.

BTW, google's cloud speech service is free for something crazy like 1m characters per month. So unless you are doing a ton of TTS it should be free. Especially if you repeat messages a lot and can take advantage of a cache.
